source/client-side-encryption/tests/README.rst

```diff
@@ -1080,3 +1161,143 @@ Invalid Hostname in KMS Certificate
    In Python, this message is "certificate verify failed: IP address mismatch, certificate is not valid for '127.0.0.1'". In Go, this message
    is "cannot validate certificate for 127.0.0.1 because it doesn't contain any IP SANs". If the language of implementation has a single, generic
    error message for all certificate validation errors, drivers may inspect other fields of the error to verify its meaning.
+
+KMS TLS Options Test
+~~~~~~~~~~~~~~~~~~~~
+
+Two mock KMS server processes must be running:
+
+1. The mock `KMS HTTP server <https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/csfle/kms_http_server.py>`_.
+
+   Run on port 8002 with `ca.pem`_ as a CA file and `server.pem`_ as a cert file.
+
+   Run with the ``--require_client_cert`` option.
+   
+   Example:
+
+   .. code::
+
+      python -u kms_http_server.py --ca_file ../x509gen/ca.pem --cert_file ../x509gen/server.pem --port 8002 --require_client_cert
+   
+
+2. The mock `KMS KMIP server <https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/csfle/kms_kmip_server.py>`_.
+
+
+Create a ``ClientEncryption`` object (referred to as ``client_encryption``)
+
+Configure with ``keyVaultNamespace`` set to ``keyvault.datakeys``, and a default MongoClient as the ``keyVaultClient``.
+
+Configure with KMS providers as follows:
+
+.. code:: javascript
+
+   {
+         "aws": {
+            "accessKeyId": <set from environment>,
+            "secretAccessKey": <set from environment>
+         },
+         "azure": {
+            "tenantId": <set from environment>,
+            "clientId": <set from environment>,
+            "clientSecret": <set from environment>,
+            "identityPlatformEndpoint": "127.0.0.1:8002"
+         },
+         "gcp": {
+            "email": <set from environment>,
+            "privateKey": <set from environment>,
+            "endpoint": "127.0.0.1:8002"
+         },
+         "kmip" {
+            "endpoint": "localhost:5698"
```

Comment:
KMIP uses "localhost". The other providers use "127.0.0.1".

`kmip` permits endpoints with no dot separators. `localhost` is valid.

The endpoints for `aws`, `azure`, and `gcp` require a dot separator. If `localhost` is used, libmongocrypt returns an error when parsing the endpoint [here](https://github.com/mongodb/libmongocrypt/blob/07655598b8478f457a252ab955a6a8299fe17575/src/mongocrypt-endpoint.c#L85:L90) with the message: "Invalid endpoint, expected dot separator in host".

Reply:
That's kind of weird, but ok.

Reply:
Updated the `kmip` endpoints to use an IP literal for consistency.